src/main/java/controllers/UserProfileController.java

```diff
@@ -87,128 +90,164 @@ else if (choice == 1) {
-        }
-        else {
+        } else {
             this.inOut.setOutput("Bio given is the same as the old one. Returning to main page");
         }
     }
 
     /**
-     * Return error msg if target_user is already followed by user, return successful msg
+     * Runs the follow user logic: display error msg if target_user is already followed by user, return successful msg
      * otherwise and follow target_user
+     *
+     * @param user the currently logged-in user
+     * @param targetUser the user that the currently logged-in user wishes to follow
      */
     private void runFollowUser(User user, User targetUser) {
         if (this.userManager.followUser(user, targetUser)) {
             this.userManager.followUser(user, targetUser);
             this.inOut.setOutput("Successfully followed the target user!");
-        }
-        else {
+        } else {
             this.inOut.setOutput("Target user already followed!");
         }
     }
 
     /**
-     * Return false if target_user is not followed by user, return true otherwise and unfollow target_user
+     * Runs the unfollow user logic
+     */
+    private void runUnfollowUser() {
+        try {
+            String usernameToUnfollow =
+                    this.inOut.getInput("Enter the username of the person you'd like to unfollow: ");
+            User userToUnfollow = this.userManager.findUser(usernameToUnfollow);
+            if (this.userManager.unfollowUser(this.loginManager.getCurrUser(), userToUnfollow)) {
+                this.inOut.setOutput("Successfully unfollowed the target user!");
+            } else {
+                this.inOut.setOutput("User is not followed to begin with!");
+            }
+        } catch (IOException e) {
+            inOut.setOutput("There was an error: " + e);
+        }
+    }
+
+    /**
+     * Display all currently registered usernames
      */
-    private void runUnfollowUser(User user, User targetUser) {
-        if (this.userManager.unfollowUser(user, targetUser)) {
-            this.inOut.setOutput("Successfully unfollowed the target user!");
+    private void runDisplayAllUsers() {
+        User[] users = this.databaseManager.getAllUsers();
+        StringBuilder allUsernames = new StringBuilder();
+        for (User user : users) {
+            allUsernames.append(" ").append(this.userManager.getUsername(user)).append(", ");
         }
-        else {
-            this.inOut.setOutput("User is not followed to begin with!");
+        this.inOut.setOutput("Currently registered users:" + allUsernames.substring(0, allUsernames.length() - 2));
+    }
+
+    /**
+     * Display user's posts
+     */
+    private void runDisplayUserPosts(User user) {
+        PostController postController = new PostController(this.inOut, this.databaseManager, this.loginManager);
+        PostManager postManager = new PostManager(this.userManager.getUserPosts(user));
+        for(Post post: this.userManager.getUserPosts(user)) {
```

Comment:
Could we add a space after "for" for style consistency?

Reply:
yeah for sure